s3torchconnector/src/s3torchconnector/s3iterable_dataset.py

```diff
@@ -133,6 +148,42 @@ def _get_transformed_object(self, bucket_key: S3BucketKeyData) -> Any:
         )
 
     def __iter__(self) -> Iterator[Any]:
-        return map(
-            self._get_transformed_object, self._get_dataset_objects(self._get_client())
+        worker_id = 0
+        num_workers = 1
+        if not self._share_dataset_within_process:
```

Comment:
nit: why not handing the if `self._share_dataset_within_process` first as the base condition and have the else implemented below? reasoning about NOT conditions are harder.